risc0/circuit/rv32im/src/prove/emu/exec/mod.rs

```diff
@@ -436,8 +436,11 @@ impl<'a, 'b, S: Syscall> Executor<'a, 'b, S> {
     fn ecall_software(&mut self) -> Result<bool> {
         tracing::trace!("[{}] ecall_software", self.insn_cycles);
         let into_guest_ptr = ByteAddr(self.load_register(REG_A0)?);
-        let into_guest_len = self.load_register(REG_A1)? as usize;
-        if into_guest_len > 0 && !is_guest_memory(into_guest_ptr.0) {
+        let into_guest_len = self.load_register(REG_A1)?;
```

Comment:
Why not cast this to `usize` here? This avoids casting noise later.

Reply:
I've removed this change.